src/doc.d

```diff
@@ -1230,9 +1239,18 @@ extern (C++) void toDocBuffer(Dsymbol s, OutBuffer* buf, Scope* sc)
             // emit constraints if declaration is a templated declaration
             if (td && td.constraint)
             {
-                buf.writestring(" if (");
+                bool funcDecl = td.isFuncDeclaration() is null;
+                if (funcDecl)
+                {
+                    buf.writestring("$(DDOC_CONSTRAINT if (");
```

Comment:
I think it's better to leave the `if (` and the closing `)` to the macro. That way folks who want to stylize it differently may do so. So here you only have `$(DDOC_CONSTRAINT ` and then the `DDOC_CONSTRAINT` macro has `if ($0)` in its definition.